src/gen_dispatch.py

```diff
@@ -501,11 +501,12 @@ def write_header(self, out_file):
                 self.outln('#endif')
                 self.outln('#endif')
         else:
+            self.outln('#ifdef HAVE_KHRPLATFORM_H')
```

Comment:
This relies on everyone using Epoxy to:

 1. check if `khrplatform.h` is available
 2. define `HAVE_KHRPLATFORM_H` as part of their build, either by:
   - injecting something like `-DHAVE_KHRPLATFORM_H` in their compiler flags *or*
   - only ever including Epoxy headers after they include a "config.h" which defines that symbol

Which is somewhat suboptimal for everything that is not a leaf application. For instance, GTK exposes `epoxy/gl.h` in its public headers, which means everyone using GTK would now have to either use the existing fallback (most likely) or do the `khrplatform.h` discovery, because people will build Epoxy on one system, and build code against Epoxy on another.

That's why the comment says: "coming up with #ifdefs to decide when it's not present on the platform would be hard". It's not hard to add a pre-processor symbol: it's hard to do so while without making every downstream use of Epoxy more complicated.

Reply:
Yes, this patch _does_ rely on `HAVE_KHRPLATFORM_H` ending up in `config.h`.

> [...] because people will build Epoxy on one system, and build code against Epoxy on another.

Oh, that's a bit of a surprise to me.  Maybe I'm missing something but doing this may introduce subtle inconsistencies, no?  If the platform on which libepoxy is built have some subtle differences in EGL APIs (types, pre-procesor macros and whatnot) to those on the platform where it's actually used (and linked against), doesn't it result in any problems?

Reply:
Any update on this progress?